### ISSUE TITLE
Execute script automatically on attach or initialize

### ISSUE THREAD

**Issue by torusrxxx** [x64dbg/x64dbg#760]
The script file can be a global file or a file specific to the debuggee. It runs when the system breakpoint or attach breakpoint is reached.

**Comment by mrfearless** [comment]
Could do this with a plugin, via CBSYSTEMBREAKPOINT and CBATTACH call _plugin_startscript for script 'registered' to a specific exe. Or would it be better to integrate this functionality directly into the x64dbg codebase?

**Comment by mrexodia** [comment]
Yeah true, it could be done with a plugin right now.

**Comment by mrfearless** [comment]
I could adjust an existing plugin i have to do this i think, but if its something to integrate directly into code there is probably no point in me doing that. Ill wait and see what direction this might go before doing anything.

**Comment by Treetippayanipa** [comment]
IssueReport

**Comment by mrfearless** [comment]
So i started creating a plugin to do this, but ran into a few small issues.

I have a test script located at: M:\x64dbg\scripts\test.txt

If i load a file to debug and use the command bar to use:

```scriptload "M:\x64dbg\scripts\test.txt"```

It loads the script fine.

When using the plugin i use the following code (after detecting if it matches a module that has a script associated to be auto executed - info is stored in an ini file for the plugin:

```Invoke DbgScriptLoad, Addr szNewScriptFilename```

szNewScriptFilename contains the string "M:\x64dbg\scripts\test.txt" (without quotes)

On an older codebase build i got _CreateFile error_ - now that the code is using the ```filehelper.cpp``` i get the _FileHelper::ReadAllText failed..._ message (ive downloaded and compiled from the most recent commit)

I modified the ```simplescript.cpp``` like so to check the string i was passing is correct:

```
static bool scriptcreatelinemap(const char* filename)
{

	char message[256] = "";
	sprintf(message, "FileHelper::ReadAllText... %s ", filename);
	GuiScriptError(0, message);

    String filedata;
    if(!FileHelper::ReadAllText(filename, filedata))
    {
		char message[256] = "";
		sprintf(message, "FileHelper::ReadAllText failed... %s ", filename);
		GuiScriptError(0, message);
        //GuiScriptError(0, "FileHelper::ReadAllText failed...");
        return false;
    }
```

So i get a messagebox beforehand (using commandbar or via plugin) to verify filename is same as the test string: "M:\x64dbg\scripts\test.txt"

commandbar version works, but plugin then comes back with the _FileHelper::ReadAllText failed_ message (i modified it slightly as well just to verify string is correct as well)

I also get an error if trying to call ```Invoke DbgScriptRun, 0``` - error says _Debugger must be paused to run a script!_

I tried using the _plugin_debugpause function but that does pause the debugger and wont continue with the other commands in the plugin if issued before DbgScriptLoad and/or DbgScriptRun

So im not sure why the CreateFile is failing to open the file. Unless it is to do with a thread that should be running (or not running) for the scriptload (DbgScriptLoad) to work.

I can upload the test plugin source and binary if required.

**Comment by torusrxxx** [comment]
In fact you cannot write a plugin to do it. When the plugin callback is invoked, the debugger is not in paused state. Something must be changed in the dbg.

**Comment by mrfearless** [comment]
Yes I thought that as well.

**Comment by mrexodia** [comment]
You can break on pause as well...

**Comment by mrexodia** [comment]
should be fixed with #1063
